yammer/yammer-authenticator/org.wso2.carbon.identity.authenticator.yammer/src/main/java/org/wso2/carbon/identity/authenticator/YammerOAuth2Authenticator.java

```diff
@@ -0,0 +1,236 @@
+                                                String callbackurl)
+            throws AuthenticationFailedException {
+        OAuthClientRequest accessRequest;
+        try {
+            accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint)
+                    .setGrantType(GrantType.AUTHORIZATION_CODE).setClientId(clientId)
+                    .setClientSecret(clientSecret).setRedirectURI(callbackurl).setCode(code)
+                    .buildBodyMessage();
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        }
+        return accessRequest;
+    }
+
+    private OAuthClientResponse getOauthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws AuthenticationFailedException {
+        OAuthClientResponse oAuthResponse = null;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        }
+        return oAuthResponse;
+    }
+
+    /**
+     * Get subject attributes.
+     *
+     * @param token                   OAuthClientResponse
+     * @param authenticatorProperties Map<String, String>
+     * @return Map<ClaimMapping, String> Claim mappings.
+     */
+    @Override
+    protected Map< ClaimMapping, String > getSubjectAttributes(OAuthClientResponse token,
+                                                               Map< String, String > authenticatorProperties) {
+        Map< ClaimMapping, String > claims = new HashMap<>();
+        try {
+            String jsonString = token.getParam("access_token");
+            String accessToken = JSONUtils.parseJSON(jsonString).get(YammerOAuth2AuthenticatorConstants.ACCESS_TOKEN).toString();
+            String url = getUserInfoEndpoint(token, authenticatorProperties);
+
+            String json = sendRequest(url, accessToken);
+            if (StringUtils.isBlank(json)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Unable to fetch user claims. Proceeding without user claims");
+                }
+                return claims;
+            }
+            Map< String, Object > jsonObject = JSONUtils.parseJSON(json);
+            for (Map.Entry< String, Object > data : jsonObject.entrySet()) {
+                String key = data.getKey();
+                claims.put(ClaimMapping.build(key, key, null, false), jsonObject.get(key).toString());
+                if (log.isDebugEnabled()) {
+                    log.debug("Adding claims from end-point data mapping : " + key + " - " +
+                            jsonObject.get(key).toString());
+                }
+            }
+        } catch (Exception e) {
```

Comment:
If possible catch the specific exception type rather than catching it from the top level